staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go

```diff
@@ -51,6 +51,7 @@ func ValidateLabelSelectorRequirement(sr metav1.LabelSelectorRequirement, fldPat
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("operator"), sr.Operator, "not a valid selector operator"))
 	}
 	allErrs = append(allErrs, ValidateLabelName(sr.Key, fldPath.Child("key"))...)
+	allErrs = append(allErrs, ValidateLabelValues(sr.Values, fldPath.Child("values"))...)
```

Comment:
Yes. this is a validation for label selector, it should validate both key & values to avoid bad consequences.